pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -169,6 +171,32 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 		return nil, err
 	}
 
+	// Replace the standard k8s emptyDir volume plugin with a wrapper version
+	// which offers XFS quota functionality, but only if the node config
+	// specifies an empty dir quota to apply to projects:
+	if options.ProjectEmptyDirQuota != nil {
+		glog.V(2).Info("Replacing empty-dir volume plugin with quota wrapper")
+		wrappedEmptyDirPlugin := false
+		for idx, plugin := range cfg.VolumePlugins {
+			// Can't really do type checking or use a constant here as they are not exported:
+			if plugin.Name() == "kubernetes.io/empty-dir" {
```

Comment:
If this goes upstream, will we patch empty dir or expose a new volume
plugin?

On Feb 25, 2016, at 4:13 PM, Dan Mace <notifications@github.com> wrote:

In pkg/cmd/server/kubernetes/node_config.go
<https://github.com/openshift/origin/pull/7351#discussion_r54165366>:

> @@ -169,6 +171,32 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
>  		return nil, err
>  	}
>
> +	// Replace the standard k8s emptyDir volume plugin with a wrapper version
> +	// which offers XFS quota functionality, but only if the node config
> +	// specifies an empty dir quota to apply to projects:
> +	if options.ProjectEmptyDirQuota != nil {
> +		glog.V(2).Info("Replacing empty-dir volume plugin with quota wrapper")
> +		wrappedEmptyDirPlugin := false
> +		for idx, plugin := range cfg.VolumePlugins {
> +			// Can't really do type checking or use a constant here as they are not exported:
> +			if plugin.Name() == "kubernetes.io/empty-dir" {

Does it really seem likely the plugin interface methods in use here are
going to change during a minor or other rebase without us noticing? I agree
that a carried patch to emptyDir is technically cleaner, I'm wondering if
there really is enough risk to warrant reworking at this point.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/7351/files#r54165366>.